crate/server/src/config/command_line/clap_config.rs

```diff
@@ -122,6 +126,61 @@ pub struct ClapConfig {
     pub non_revocable_key_id: Option<Vec<String>>,
 }
 
+impl ClapConfig {
+    /// # Errors
+    /// Fails if the configuration file is not found or if the configuration file is not valid
+    /// or if the configuration file cannot be read
+    /// or if the configuration file cannot be parsed
+    /// or if the configuration file is not a valid toml file
+    #[allow(clippy::print_stdout)] // Logging is not being initialized yet, just use standard prints
+    pub fn load_from_file() -> KResult<Self> {
+        let conf = std::env::var("COSMIAN_KMS_CONF").map_or_else(
+            |_| PathBuf::from(KMS_SERVER_CONF),
+            |conf_path| {
+                let conf_path = PathBuf::from(conf_path);
+                if conf_path.exists() {
+                    conf_path
+                } else {
+                    println!(
+                        "WARNING: Configuration file {conf_path:?} not found. Fallback to the \
+                         default path: {KMS_SERVER_CONF}"
+                    );
+                    // fallback to the default path
+                    PathBuf::from(KMS_SERVER_CONF)
+                }
+            },
+        );
+
+        let clap_config = if conf.exists() {
+            drop(Self::parse()); // Do that do catch --help or --version even if we use a conf file
+
+            println!(
+                "Configuration file {conf:?} found. Command line arguments and env variables are \
+                 ignored."
+            );
+
+            let conf_content = std::fs::read_to_string(&conf).map_err(|e| {
+                KmsError::ServerError(format!(
+                    "Cannot read KMS server config at: {conf:?} - {e:?}"
+                ))
+            })?;
+            toml::from_str(&conf_content).map_err(|e| {
+                KmsError::ServerError(format!(
+                    "Cannot parse kms server config at: {conf:?} - {e:?}"
```

Comment:
`ERROR: Cannot parse kms server config at: {conf:?} - {e:?}`

Reply:
the `ServerError` already has this prefix: `Unexpected server error:`. Is it enough?

Reply:
ok